lib/evmone/eof.cpp

```diff
@@ -341,18 +342,23 @@ std::variant<EOFValidationError, int32_t> validate_max_stack_height(
         auto stack_height_required = instr::traits[opcode].stack_height_required;
         auto stack_height_change = instr::traits[opcode].stack_height_change;
 
+        auto stack_height = stack_heights[i];
+        assert(stack_height != LOC_UNVISITED);
+
         if (opcode == OP_CALLF)
         {
             const auto fid = read_uint16_be(&code[i + 1]);
 
             stack_height_required = static_cast<int8_t>(code_types[fid].inputs);
+
+            if (stack_height + code_types[fid].max_stack_height - stack_height_required >
```

Comment:
In case of stack underflow this expression can be negative, I think it's ok, it's signed and won't underflow

Reply:
I haven't found any tests for stack underflow validation at all. I will add some at least for CALLF.

Reply:
I was wrong it cannot be negative, because callee's `max_stack_height >= inputs`